### PR TITLE
Tests: Add a test for fill to WorldTestBase

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
A unit test to WorldTestBase to test that worlds successfully generate, fulfill the accessibility rules, and there are no unplaced items from the itempool.

## How was this tested?
Tests. Current failure is a Minecraft generation crash with the options of that test base.

### CODE REVIEW

**Comment by black-sliver** [comment]
Can you explain what it does and why it fails for minecraft and, if it's something kono should fix, ping kono?

**Comment by alwaysintreble** [comment]
Generates a full solo game, calling every step that gets done normally through Main up through post_fill for a solo world, other than locality rules, progression balancing, and item_links for the provided WorldTestBase. It then loops through all the locations, until the game is either beatable or all locations have been reached. If it can't reach any locations in a loop (sphere), yet locations still exist, that means it can't reach them so it fails. If it goes through all the locations and with all of them reachable and it can't beat the game still, it fails. This part is similar to the all_state test, but it essentially checks the playthrough. Minecraft is failing during fill itself which is actually not part of the test, but this did find it. Reported it here: https://discord.com/channels/731205301247803413/1134655045930061874/1134655045930061874. The only thing this test is really "missing" is that it will still test the generated world if it has minimal, and that will pass even if it has unreachable locations, if it can still reach its goal. I'm not sure if this test should explicitly test that case or not, since that's technically intended behavior.

**Comment by alwaysintreble** [comment]
Updated it to now check all locations in the entire multiworld. If there are still locations remaining that can't be reached, and accessibility isn't minimal it'll fail, and if accessibility is minimal and there's locations it can't reach, but it can complete the game, it'll pass. I *think* this is what we want. I debugged my accessibility testing classes in Messenger to check it behaves how I'd expect.

**Comment by alwaysintreble** [comment]
This PR now includes an option change in minecraft to fulfill the test, and a fix for a bad import on current main

File: worlds/minecraft/Options.py [hunk 1]
```diff
@@ -14,15 +14,15 @@ class EggShardsRequired(Range):
...
+    range_end = 50
     default = 0
 
 
 class EggShardsAvailable(Range):
     """Number of dragon egg shards available to collect."""
     display_name = "Egg Shards Available"
     range_start = 0
-    range_end = 74
+    range_end = 50
```
Comment (KonoTyran) [inline worlds/minecraft/Options.py:25]: So @espeon65536 if I understand this correctly this is the lowest max possible if both bosses are turned on, exclude post-game, and all hard/unreasonable are turned off as well?

Basically the limit of free slots after using settings to exclude the maximum number of checks?

Comment (espeon65536) [reply worlds/minecraft/Options.py:25]: That's correct. (It's actually like 52, but I think 50 is a nicer number + it prevents some possible problems where all items are progression.)

**Review by black-sliver — APPROVED** [review]

**Pull request merged**